app/controllers/foreman_tasks/tasks_controller.rb

```diff
@@ -12,26 +12,15 @@ def show
 
     def index
       params[:order] ||= 'started_at DESC'
-      respond_to do |format|
-        format.html do
-          @tasks = filter(resource_base)
-          render :index, layout: !request.xhr?
-        end
-        format.csv do
-          @tasks = filter(resource_base, paginate: false)
-          csv_response(@tasks, [:id, :action, :state, :result, 'started_at.in_time_zone', 'ended_at.in_time_zone', :username], ['Id', 'Action', 'State', 'Result', 'Started At', 'Ended At', 'User'])
-        end
-      end
+      process_index resource_base
     end
 
     def summary
       render json: Task::Summarizer.new(resource_base, params[:recent_timeframe].to_i).summary
     end
 
     def sub_tasks
-      task   = resource_base.find(params[:id])
-      @tasks = filter(task.sub_tasks)
-      render :index
+      process_index resource_base.find(params[:id]).sub_tasks
```

Comment:
don't we lose taxonomies filtering, pagination (may be wanted) and searching on sub_tasks with this?

Reply:
No, the filtering is done inside ~~`process_index`~~ `respond_with_tasks` and seems to work fine.